index.js

```diff
@@ -51,8 +55,17 @@ function propertyIsUnsafe(target, key) {
 			&& Object.propertyIsEnumerable.call(target, key)) // and also unsafe if they're nonenumerable.
 }
 
+// Retrieves either a new object or the appropriate target object to mutate.
+function getDestinationObject(target, options) {
+	if (options && options.mergeWithTarget) {
```

Comment:
@RebeccaStevens does this need to account for `options.clone` scenarios for objects that are not the root target? I've read through it a few times and with `deepmerge` being self-referenced, I wonder if there's a case where this function would be called in reference to an object nested deeper than the entrypoint where we might want to return the target here in the case where `options.clone` is `false`?

Reply:
No I don't think so. This function is only called at most once per call of the main deepmerge function.

Reply:
Per my last commit, this is called repeatedly during recursion down an object tree. Not sure if this changes your thought on the topic.

Reply:
Specifically, `deepmerge` is returned from `cloneUnlessOtherwiseSpecified` which can call `mergeObject` which calls `getDestinationObject`.

Reply:
I think I'm seeing the issue you are talking about here.

I think with `options.mergeWithTarget: true` & `options.clone: false`, merging with each target going down should happen.
I think with `options.mergeWithTarget: true` & `options.clone: true`, merging should only happen with the top most target.

Reply:
Looking at the codebase, it doesn't really look set up to be able to add this functionality easily in a non-hacky way.
A bunch of refactoring may be needed (which I'm in the process of doing in #215).

If you want to just do it in a hacky way, you could just remove `mergeWithTarget` from the options object before it is passed to the recursive call.